dashboard/app/controllers/api/v1/pd/applications_controller.rb

```diff
@@ -58,11 +59,18 @@ def quick_view
 
   # PATCH /api/v1/pd/applications/1
   def update
-    if application_params[:response_scores]
-      JSON.parse(application_params[:response_scores]).transform_keys! {|x| x.to_s.underscore}.to_json
+    application_data = application_params.except(:locked)
+
+    if application_data[:response_scores]
+      JSON.parse(application_data[:response_scores]).transform_keys! {|x| x.to_s.underscore}.to_json
```

Comment:
I don't think you want the bang function `transform_keys!` here. See http://api.rubyonrails.org/classes/Hash.html#method-i-transform_keys-21. `transform_keys!` modified the source hash, which in this case is the result of `JSON.parse(application_data[:response_scores])`, a throwaway. Use the functional version of `tranform_keys` instead.